superset/connectors/druid/models.py

```diff
@@ -608,17 +614,13 @@ def lookup_datasource(d: DruidDatasource) -> Optional[DruidDatasource]:
                 db.session.query(DruidDatasource)
                 .filter(
                     DruidDatasource.datasource_name == d.datasource_name,
-                    DruidCluster.cluster_name == d.cluster_name,
+                    DruidDatasource.cluster_id == d.cluster_id,
                 )
                 .first()
             )
 
         def lookup_cluster(d: DruidDatasource) -> Optional[DruidCluster]:
-            return (
-                db.session.query(DruidCluster)
-                .filter_by(cluster_name=d.cluster_name)
-                .one()
-            )
+            return db.session.query(DruidCluster).filter_by(id=d.cluster_id).one()
```

Comment:
Somewhat unrelated to this PR but the return type here is `Optional[DruidCluster]` whereas the call is to `.one()` (which could throw) as opposed to `.one_or_none()`.